File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/MessageApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class MessageApi {
    private final HttpClient client;
    
    public MessageApi(HttpClient client) {
        this.client = client;
    }

    /** Count Claude tokens */
    public CountClaudeTokensPostResponse countClaudeTokens(CountClaudeTokensPostRequest body) throws Exception {
        return (CountClaudeTokensPostResponse) client.post(ApiPaths.aiPath("/messages/count_tokens"), body);
    }

    /** Create Claude message */
    public CreateClaudeMessageResponse createClaude(CreateClaudeMessageRequest body, Map<String, String> headers) throws Exception {
        return (CreateClaudeMessageResponse) client.post(ApiPaths.aiPath("/messages"), body, null, headers);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/MessageApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class MessageApi(private val client: HttpClient) {

    /** Count Claude tokens */
    suspend fun countClaudeTokens(body: CountClaudeTokensPostRequest): CountClaudeTokensPostResponse? {
        return client.post(ApiPaths.aiPath("/messages/count_tokens"), body) as? CountClaudeTokensPostResponse
    }

    /** Create Claude message */
    suspend fun createClaude(body: CreateClaudeMessageRequest, headers: Map<String, String>? = null): CreateClaudeMessageResponse? {
        return client.post(ApiPaths.aiPath("/messages"), body, null, headers) as? CreateClaudeMessageResponse
    }
}
